### PR TITLE
Generate Commands code

### PR DESCRIPTION
**Pull request by ralucabrie**

### CODE REVIEW

File: packages/generators/cypress/src/code.ts [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+      `;
+      fs.appendFileSync(selectorFile, fileContent);
+    }
+  }
+
+  private createGroupFolders(group: string) {
+    const folderGroup: string = group;
+    let basePath = this.localTestFolder;
+    const folderArr: string[] = [...folderGroup.split('/')];
+    for (const folder of folderArr) {
```
Comment (cstefanache) [inline packages/generators/cypress/src/code.ts:279]: is this used just for a single start -> branch end?
if not - line 283 (where the current folder gets appended to the base path) will not work for multi levels
```
level 1
   level 1.1
      level 1.1.1
   level 1.2
      level 1.2.1
```

File: packages/generators/cypress/src/code.ts [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+      if (!fs.existsSync(`${basePath}/${folder}`)) {
+        fs.mkdirSync(`${basePath}/${folder}`);
+      }
+      basePath = `${basePath}/${folder}`;
+    }
+  }
+
+  private getVariableName(p: string) {
+    if (p.includes('${store.')) {
```
Comment (cstefanache) [inline packages/generators/cypress/src/code.ts:288]: I think that this works if the formula has a single variable `${store.varName}`. Please check if it works for `${stor.varName1} + ${store.varName2}`

File: packages/generators/cypress/src/code.ts [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+      const { group, specs } = functionality;
+      const groupFolder = `${this.localTestFolder}${group}`;
+      this.createGroupFolders(group);
+      for (const spec of specs) {
+        const { file, beforeAll, tests } = spec;
+        const specFile = `${groupFolder}/${file}`;
+        let startDecribeCommand = `
+        describe('${file}', () => {
+          let store={};
+          let library = {
```
Comment (cstefanache) [inline packages/generators/cypress/src/code.ts:367]: create a new issue: `load library file as text from the discovery package and inject in code`

File: packages/generators/cypress/src/structure.ts [hunk 4]
```diff
@@ -2,24 +2,36 @@ import * as fs from 'fs';
...
+  MethodDefinition,
+  Selector,
 } from './models/models';
-import { ConfigModel } from './models/config';
+import { ConfigModel, Iterator } from './models/config';
 export default class Structure {
   rawFlow: FlowModel;
   rawConfig: Config;
-  inputPath: string = 'output/marvin2';
+  inputPath: string = '/Users/ralucabrie/Desktop/CodeGen';
```
Comment (cstefanache) [inline packages/generators/cypress/src/structure.ts]: hardcoded local folder

File: packages/generators/cypress/src/structure.ts [hunk 5]
```diff
@@ -2,24 +2,36 @@ import * as fs from 'fs';
...
   flow: NewFlowModel = {
     functionalities: [],
     selectors: [],
     commands: [],
   };
-  config: ConfigModel;
+  config: ConfigModel = {
```
Comment (cstefanache) [inline packages/generators/cypress/src/structure.ts:24]: extract models in separate file

File: packages/generators/cypress/src/structure.ts [hunk 6]
```diff
@@ -52,18 +64,57 @@ export default class Structure {
       .replace(new RegExp('[ :/\\\\]', 'g'), '-');
```
Comment (cstefanache) [inline packages/generators/cypress/src/structure.ts:64]: would love to see all regular expressions extracted in a separate file with comments

File: packages/generators/cypress/src/structure.ts [hunk 7]
```diff
@@ -138,110 +189,175 @@ export default class Structure {
...
-    return sequence.find((step: any) => step.store === 'true') ? true : false;
+    return sequence.find((step: any) => step.store === true) ? true : false;
   }
 
   private getParameters(sequence: any) {
     const params: Identifier[] = [];
     for (const step of sequence) {
-      if (step.type === 'clearAndFill' || step.type === 'fill') {
+      if (
+        step.type === 'clearAndFill' ||
```
Comment (cstefanache) [inline packages/generators/cypress/src/structure.ts:232]: `clearAndFill`, `fill`, `click` can be extracted as constants

File: packages/generators/cypress/src/structure.ts [hunk 8]
```diff
@@ -138,110 +189,175 @@ export default class Structure {
...
+        siblings: [...this.getSiblings(iterator)],
+      });
     }
+    return array;
+  }
 
+  public generate() {
+    console.log('+++++++++++++++++++++++++++++++++++++++++++++++++++++++++');
+    this.flow.functionalities = this.getFunctionalities(this.rawFlow.graph);
+    // for (const func of this.flow.functionalities) {
```
Comment (cstefanache) [inline packages/generators/cypress/src/structure.ts:320]: remove commented code

File: packages/discovery/src/runner.ts [hunk 9]
```diff
@@ -135,6 +133,25 @@ export default class Runner {
...
+        if (element) {
+          const value = await page.evaluate(
+            (element: any) => element.getAttribute('value'),
+            element
+          );
+          const text = await element.evaluate((el: any) =>
+            el.textContent?.trim()
+          );
+
+          log(`+++++++++++ ${value}, ${text}`);
```
[the remark this answers is not in the document]
Comment (ralucabrie) [reply packages/discovery/src/runner.ts:148]: done

File: packages/generators/cypress/src/code.ts [hunk 10]
```diff
@@ -0,0 +1,395 @@
...
+  private localSupportFolder: string;
+  private localTestFolder: string;
+  private locatorKeyWord = 'locators';
+  private iteratorValueKewWord = 'iteratorValue';
+  private storeKeyWord = 'store';
+
+  constructor(private flow: NewFlowModel, private config: ConfigModel) {
+    this.localSupportFolder = path.join(
+      process.cwd(),
+      '/packages/sample-frontend-e2e/src/support'
```
[the remark this answers is not in the document]
Comment (ralucabrie) [reply packages/generators/cypress/src/code.ts:27]: done

File: packages/generators/cypress/src/code.ts [hunk 11]
```diff
@@ -0,0 +1,395 @@
...
+    );
+    this.localTestFolder = path.join(
+      process.cwd(),
+      '/packages/sample-frontend-e2e/src/e2e'
```
[the remark this answers is not in the document]
Comment (ralucabrie) [reply packages/generators/cypress/src/code.ts:31]: done

File: packages/generators/cypress/src/structure.ts (hunk 4 above)
Comment (ralucabrie) [reply packages/generators/cypress/src/structure.ts]: done

File: packages/generators/cypress/src/structure.ts (hunk 8 above)
Comment (ralucabrie) [reply packages/generators/cypress/src/structure.ts:319]: done

File: packages/generators/cypress/src/structure.ts (hunk 6 above)
Comment (ralucabrie) [reply packages/generators/cypress/src/structure.ts:64]: done

**Pull request merged**